internal/models/flow_state.go

```diff
@@ -152,5 +153,17 @@ func (f *FlowState) VerifyPKCE(codeVerifier string) error {
 }
 
 func (f *FlowState) IsExpired(expiryDuration time.Duration) bool {
+	if f.AuthenticationMethod == MagicLink.String() {
+		return time.Now().After(f.IssuedAt.Add(expiryDuration))
```

Comment:
I don't think this checks that `f.IssuedAt` can be null, causing a panic.

What happens to the magic links already in the system, that are still valid?

Reply:
Magic links already in the system would obtain an `IssuedAt` when the link is clicked.

There would be an issue for Magic Links which have been clicked but whose associated Auth Codes have not been consumed. I'll add an additional check there - thanks!